src/cellrank/estimators/terminal_states/_gpcca.py

```diff
@@ -532,6 +536,165 @@ def set_initial_states(
         )
         return self
 
+    def tsi(
```

Comment:
@michalk8, I refactored the function quite a bit:

1. It's called `tsi` now. I think it fits the remaining code base better since it is a function that is called by the user; we don't call methods `get_XXX`, for example, `compute_fate_probabilities` instead of `get_fate_probabilities`,  anywhere else as far as I am aware.
2. The arguments `terminal_states` and `cluster_key` are optional and only needed if the TSI needs to be (re)computed.
3. The TSI is saved as an AnnData object to check if it needs to be recomputed when passing a new set of terminal states or cluster key.

Let me know what you think.

Reply:
Yes, I thought the same, but I don't think the user needs access to the AnnData instance so I decided against it.